### PR TITLE
Update building-and-testing-ruby.md

### PR DESCRIPTION
**Pull request by qhartman**
### Why:

Changing to use action version v1 as recommended at https://github.com/ruby/setup-ruby so that current versions of ruby will be supported rather than falling out of date by pinning to a particular hash.

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Just updating the pinned version of the action
### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline.

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by qhartman** [comment]
Was not able to review in the preview env as the deploy there seems to have failed, presumably I don't have permissions to do that. Quite a small change though, so I'm confident it's correct.

**Comment by cmwilson21** [comment]
@qhartman Thanks so much for opening a PR! I'll get this triaged for review :zap:

And welcome to the community!

**Comment by qhartman** [comment]
OK, I was able to verify my change at https://docs-26345-268525.preview.ghdocs.com/en/actions/automating-builds-and-tests/building-and-testing-ruby once you got the deploy through. Thanks @cmwilson21 , I think this is ready to go once it has approval.

**Comment by cmwilson21** [comment]
@qhartman Thanks for verifying! It's up on the board waiting review 👀 👍

**Review by SiaraMist — COMMENTED** [review]
Hey @qhartman, thank you for opening this PR! ⚡ 

I understand why you updated this to use the version instead of a specific commit SHA, automatically getting updates for Actions is definitely more convenient. However, [according to our style guide](https://docs.github.com/en/contributing/writing-for-github-docs/style-guide#pinning-version-numbers-to-sha), for non-GitHub Actions we pin to a specific SHA to align with [our security advice](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions). I left a suggestion to update this to a newer SHA instead of pinning the action to the version. If that looks good to you, feel free to accept the change, and then I will get this merged!

Thank you again for contributing to GitHub Docs! 💛

File: content/actions/automating-builds-and-tests/building-and-testing-ruby.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ jobs:
     steps:
       - uses: {% data reusables.actions.action-checkout %}
       - name: Set up Ruby
-        uses: ruby/setup-ruby@ec02537da5712d66d4d50a0f33b7eb52773b5ed1
+        uses: ruby/setup-ruby@v1
```
[the remark this answers is not in the document]
Comment (Miguel736j) [reply content/actions/automating-builds-and-tests/building-and-testing-ruby.md:58]: 59 59

**Comment by qhartman** [comment]
@SiaraMist Thanks for the review and suggestion. It seems your example and style-guide recommendations are in conflict with those of the authors of the Ruby Action, per the note at https://github.com/ruby/setup-ruby. Perhaps an additional note here with the caveats they mention and/or at least a callout that they make a different recommendation would be helpful for newcomers.

**Comment by SiaraMist** [comment]
That makes sense, @qhartman. Going back to read the article, I do notice there is a [code comment in the example action](https://github.com/github/docs/pull/26345/files#diff-f190743620873640ce87e6eae2f5fd5402f710894eec479fba4839970a387561L40) that describes this. I linked to the specific line in your PR which isn't actually readable, but you can see the comment displayed [in the example in this section](https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-ruby#using-the-ruby-starter-workflow).

In light of that comment and the instructions to update the commit SHA, I'm actually going to close this PR instead of updating the SHA in each example on that page so they all match. 👍

**Pull request closed without merging**